der/src/asn1/any.rs

```diff
@@ -282,3 +273,70 @@ impl Tagged for Any {
         self.tag
     }
 }
+
+#[cfg(feature = "alloc")]
+impl ValueOrd for Any {
+    fn value_cmp(&self, other: &Self) -> Result<Ordering> {
+        self.value.der_cmp(&other.value)
+    }
+}
+
+#[cfg(feature = "alloc")]
+impl<'a, T> From<T> for Any
+where
+    T: Into<AnyRef<'a>>,
+{
+    fn from(input: T) -> Any {
+        let anyref: AnyRef<'a> = input.into();
+        Self {
+            tag: anyref.tag(),
+            value: ByteVec::new(anyref.value()).expect("invalid ANY"),
+        }
+    }
+}
+
+/// Internal macro to factorize implementation of [`Any`] and [`AnyRef`] types.
+macro_rules! impl_any_body {
+    ($comment:literal, $fnname:ident, $rettype:ty) => {
+        impl<'a> AnyRef<'a> {
+            #[doc = $comment]
+            pub fn $fnname(self) -> Result<$rettype> {
+                self.try_into()
+            }
+        }
+
+        #[cfg(feature = "alloc")]
+        impl Any {
+            #[doc = $comment]
+            pub fn $fnname<'a>(&'a self) -> Result<$rettype> {
+                self.try_into()
+            }
+        }
+    };
+}
+
+impl_any_body!(
+    "Attempt to decode an ASN.1 `PrintableString`.",
+    printable_string,
+    PrintableStringRef<'a>
+);
+impl_any_body!(
+    "Attempt to decode an ASN.1 `TeletexString`.",
+    teletex_string,
+    TeletexStringRef<'a>
+);
+impl_any_body!(
+    "Attempt to decode an ASN.1 `VideotexString`.",
+    videotex_string,
+    VideotexStringRef<'a>
+);
+impl_any_body!(
+    "Attempt to decode an ASN.1 `UTF8String`.",
+    utf8_string,
+    Utf8StringRef<'a>
+);
+impl_any_body!(
+    "Attempt to decode an ASN.1 `IA5String`.",
+    ia5_string,
+    Ia5StringRef<'a>
+);
```

Comment:
Not sure using macros here is actually a net win

Reply:
I'm not convinced either. I kind of wanted to express "those two needs similar implementation".
It could be a trait.
Be happy to take suggestions.

Reply:
@tarcieri 
got an alternative in https://github.com/RustCrypto/formats/pull/771/commits/b0fcddaba0349bf185977571154521a7d4cad893 let me know what you think

I guess at somepoint the signature is going to differ? As I'll want:
  - `AnyRef::utf8_string -> Utf8StringRef` 
  - `Any::utf8_string -> Utf8String` (owned)
  
  I have to experiment a bit.